### DISCUSSION TITLE
[V5] Cron Job

### DISCUSSION THREAD

**Discussion by remidosol** [adonisjs/core#2033] [Help] [answered]
Hello folks! I need to run some kind of cron job. 

Is there any support for AdonisJS V5 for **node-schedule** or **node-cron** packages? Can I use one of these packages directly with TypeScript? 

By the way, which package do you suggest to use or prefer to use, why?

**Comment by ruanitto** [comment]
I am using [adonis5-schedule](https://www.npmjs.com/package/adonis5-scheduler) that meets all my requirements.

**Comment by ruanitto** [comment, marked as answer]
For automatic execution, I am calling the service execution as soon as the application is ready within providers/AppProvider.ts

```javascript
  public async ready() {
    // App is ready
    const scheduler = this.app.container.use('Adonis/Addons/Scheduler')
    scheduler.run()
  }
```

**Reply by remidosol** [reply]
I think, creating a preload file (prldfile) going to be more useful than AppProvider but thanks.

**Reply by ErnaniR** [reply]
adonis5-scheduler explains very little in its docs... like how to start the scheduler when running the app :D

**Reply by pixelpaulaus** [reply]
did you find out how to start Scheduler via a preload file?

**Reply by ruanitto** [reply]
> did you find out how to start Scheduler via a preload file?
@pixelpaulaus create a preload file:

```sh
node ace make:preload schedules
```

Then, in start/schedules.ts, puth this content:

```typescript
import Application from '@ioc:Adonis/Core/Application'

const Scheduler = Application.container.use('Adonis/Addons/Scheduler')
Scheduler.run()
```
